File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Login/LogoutManager.java
package kore.botssdk.audiocodes.webrtcclient.Login;

import android.app.Activity;
import android.content.Context;

import kore.botssdk.audiocodes.webrtcclient.Callbacks.CallBackHandler;
import kore.botssdk.audiocodes.webrtcclient.General.ACManager;
import kore.botssdk.audiocodes.webrtcclient.General.Log;
import kore.botssdk.audiocodes.webrtcclient.General.NotificationUtils;

public class LogoutManager {
    private static final String TAG = "LogoutManager";
    private static final int LOGOUT_TIMEOUT_INTERVAL = 1;
    static Thread closeAppThread;

    public static void closeApplication(Context context) {
        Log.d(TAG, "closeApplication");
        Log.d(TAG, "close GUI");

        LoginManager.setAppState(context, LoginManager.AppLoginState.CLOSED);

//        if(BotApplication.getCurrentActivity()!=null)
//        {
//            Log.d(TAG, "close Activity");
        ((Activity) context).finish();
        ;
//        }
//        else
//        {
//            if(BotApplication.getPreviousActivity()!=null)
//            {
//                Log.d(TAG, "close prev Activity");
//        BotApplication.getPreviousActivity().finish();
        ;
//            }
//        }

        if (ACManager.getInstance().isRegisterState()) {
            Log.d(TAG, "Unregister client");
            ACManager.getInstance().startLogout();
        }

        closeAppThread = new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    Thread.sleep(LOGOUT_TIMEOUT_INTERVAL);
                } catch (InterruptedException e) {
                    Log.d(TAG, "closeAppThread interapted");
                }
                endCloseApplication(context);
            }
        });
        closeAppThread.start();
    }

    CallBackHandler.LoginStateChanged loginStateChanged = new CallBackHandler.LoginStateChanged() {
        @Override
        public void loginStateChange(Context context, boolean state) {
            if (!state) {
                if (closeAppThread != null) {
                    closeAppThread.interrupt();
                } else {
                    endCloseApplication(context);
                }
            }
            //Toast.makeText(BotApplication.getGlobalContext(), "login state: "+state, Toast.LENGTH_SHORT).show();
            //endCloseApplication();
        }
    };

    public static void endCloseApplication(Context context) {
        NotificationUtils.removeAllNotifications(context);
        Log.d(TAG, "close Process");
        android.os.Process.killProcess(android.os.Process.myPid());
    }

}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/login/LogoutManager.kt
package com.kore.ui.audiocodes.webrtcclient.login

import android.content.Context
import android.os.Process
import android.util.Log
import com.kore.ui.audiocodes.webrtcclient.general.ACManager
import com.kore.ui.audiocodes.webrtcclient.general.NotificationUtils.removeAllNotifications

class LogoutManager {
    companion object {
        private const val TAG = "LogoutManager"
        private const val LOGOUT_TIMEOUT_INTERVAL = 1
        var closeAppThread: Thread? = null
        fun closeApplication(context: Context, acManager: ACManager) {
            Log.d(TAG, "closeApplication")
            Log.d(TAG, "close GUI")
            LoginManager.setAppState(context, LoginManager.AppLoginState.CLOSED)

//        if(BotApplication.Companion.applicationContext()!=null)
//        {
//            Log.d(TAG, "close Activity");
//            BotApplication.Companion.applicationContext().finish();;
//        }
//        else
//        {
//            if(BotApplication.getPreviousActivity()!=null)
//            {
//                Log.d(TAG, "close prev Activity");
//                BotApplication.getPreviousActivity().finish();;
//            }
//        }
            if (acManager.isRegisterState) {
                Log.d(TAG, "Unregister client")
                acManager.startLogout()
            }
            closeAppThread = Thread {
                try {
                    Thread.sleep(LOGOUT_TIMEOUT_INTERVAL.toLong())
                } catch (e: InterruptedException) {
                    Log.d(TAG, "closeAppThread interapted")
                }
                endCloseApplication(context)
            }
            closeAppThread!!.start()
        }

        private fun endCloseApplication(context: Context) {
            removeAllNotifications(context)
            Log.d(TAG, "close Process")
            Process.killProcess(Process.myPid())
        }
    }
}
